VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -1,132 +1,126 @@
 package org.edx.mobile.module.notification;
 
 import android.app.Notification;
-import android.app.NotificationManager;
 import android.app.PendingIntent;
 import android.content.Context;
 import android.content.Intent;
-import android.media.RingtoneManager;
-import android.net.Uri;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
 import android.support.v4.app.NotificationCompat;
 
+import com.parse.ParseAnalytics;
 import com.parse.ParsePushBroadcastReceiver;
 
 import org.edx.mobile.R;
-import org.edx.mobile.event.CourseAnnouncementEvent;
-import org.edx.mobile.logger.Logger;
-import org.edx.mobile.util.ResourceUtil;
-import org.edx.mobile.view.CourseDetailTabActivity;
+import org.edx.mobile.base.MainApplication;
+import org.edx.mobile.module.analytics.ISegment;
+import org.edx.mobile.view.MyCoursesListActivity;
 import org.edx.mobile.view.Router;
 
-import de.greenrobot.event.EventBus;
+import java.util.Random;
 
 /**
  * subclass ParsePushBroadcastReceiver to provide fine control of
  * app's behavior on receiving the notification message
  */
 public class EdxParsePushBroadcastReceiver extends ParsePushBroadcastReceiver {
-    protected final Logger logger = new Logger(getClass().getName());
-
-    //TODO - we wont enable fly-in window for remote notification for now.
-    //UI and animation time need to get from production team.
-    private boolean enableFlyinWindow = false;
-
-    public void onReceive(android.content.Context context, android.content.Intent intent) {
-        super.onReceive(context,intent);
-     }
-
-
-    protected void onPushReceive(android.content.Context context, android.content.Intent intent) {
-        try{
-            CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
-            if ( payload != null && ParseHandleHelper.hasNotificationHash(context, payload.getIdentifier()) )
-                return;
-            if( enableFlyinWindow ) {
-                Intent intent2 = new Intent(context, NotificationFlyinService.class);
-                intent2.putExtra("com.parse.Data", intent.getExtras().getString("com.parse.Data"));
-                context.startService(intent2);
-            }
-        }catch (Exception ex){
-            logger.debug(ex.toString());
-        }
-        super.onPushReceive(context, intent);
-    }
 
+    private static ISegment getSegment(Context context) {
+        return ((MainApplication) context.getApplicationContext()).getInjector().getInstance(ISegment.class);
```

Comment:
Note that you can can inject member variables by calling `RoboGuice.injectMembers(context, this);` from the `onReceive()` callback method before calling though to the super implementation.